src/contained-list/contained-list.component.ts

```diff
@@ -0,0 +1,145 @@
+import {
+	ChangeDetectionStrategy,
+	Component,
+	HostBinding,
+	Input,
+	TemplateRef
+} from "@angular/core";
+import { ContainedListKind, ContainedListSize } from "./contained-list.enums";
+
+@Component({
+	selector: "cds-contained-list, ibm-contained-list",
+	template: `
+		<div
+			class="cds--contained-list"
+			[ngClass]="{
+				'cds--contained-list--inset-rulers': isInset,
+				'cds--contained-list--on-page': kind === ContainedListKind.OnPage,
+				'cds--contained-list--disclosed': kind === ContainedListKind.Disclosed,
+				'cds--contained-list--sm': size === ContainedListSize.Small,
+				'cds--contained-list--md': size === ContainedListSize.Medium,
+				'cds--contained-list--lg': size === ContainedListSize.Large,
+				'cds--contained-list--xl': size === ContainedListSize.ExtraLarge
+			}">
+			<div class="cds--contained-list__header">
+				<div [id]="labelId" class="cds--contained-list__label">
+					<ng-container *ngIf="!isTemplate(label)">{{ label }}</ng-container>
+					<ng-template *ngIf="isTemplate(label)" [ngTemplateOutlet]="label"></ng-template>
+				</div>
+
+				<div class="cds--contained-list__action" *ngIf="action">
+					<ng-template [ngTemplateOutlet]="action"></ng-template>
+				</div>
+			</div>
+			<div role="list" [attr.aria-labelledby]="labelId">
+				<ng-content></ng-content>
+			</div>
+		</div>
+	`,
+	styles: [`
+		:host {
+			display: block;
+		}
+	`],
```

Comment:
```suggestion
```

Don't need this since we will be assigning the classes to the div instead of the host.